plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextJavaValidator.java

```diff
@@ -451,13 +451,15 @@ public void checkValueOfNoEvent(EventValueReferenceExpression exp) {
 		} else if (eventExpr instanceof FeatureCall) {
 			element = ((FeatureCall) eventExpr).getFeature();
 		}
-
+		
 		if (element != null && (!(element instanceof Event))) {
-			String elementName = "";
+			String msg;
 			if (element instanceof NamedElement) {
-				elementName = "'" + ((NamedElement) element).getName() + "' ";
+				msg = "'" +((NamedElement) element).getName()+ "' is no event.";
+			} else {
+				msg = "Could not find event declaration." ;
 			}
-			error(elementName + "is no event.", StextPackage.Literals.EVENT_VALUE_REFERENCE_EXPRESSION__VALUE, 0,
+			error(msg , StextPackage.Literals.EVENT_VALUE_REFERENCE_EXPRESSION__VALUE, 0,
```

Comment:
is this null safe? (msg)

Reply:
yes, because in every branch of the if block, msg receives a value. I could refactor this maybe into

```
String msg = "Could not find event declaration.";
if (element instanceof NamedElement) {
    msg = "'" +((NamedElement) element).getName()+ "' is no event."; 
}
error(msg, ...);
```